runtime/src/bank.rs

```diff
@@ -7881,6 +7884,14 @@ impl Bank {
     pub fn update_accounts_hash_for_tests(&self) -> AccountsHash {
         self.update_accounts_hash(CalcAccountsHashDataSource::IndexForTests, false, false)
     }
+
+    pub fn new_program_cache_for_tx_batch_for_slot(&self, slot: Slot) -> LoadedProgramsForTxBatch {
```

Comment:
Nit: comment here that says it should not be used for TX processing, only tooling and testing.

Reply:
It's implicit as it's in the `impl` block guarded by `dev-context-utils-only` feature.